Obtaining data

In the next step , I will clean the data and structurize data. 

# 01 - Data Access and Selection

## What this notebook does
This notebook implements Stage 1 (Data Access) and Stage 2 
(Selection/Sampling) of the research workflow.

## Data used
- Collection: Metropolitan Museum of Art Open Access
- URL: https://github.com/metmuseum/openaccess  
- License: CC0 — No Rights Reserved
- Date of download: 2026-05-06

## Filters applied
- Departments selected: Asian Art, Islamic Art, Ancient Near Eastern Art
- Reason: directly relevant to research question about cultural 
  distribution across Asia and the Islamic world
- Dropped 9 redundant columns to reduce file size

## What is saved where
- Output file: ../data/asian_islamic_neareastern_met.csv
- Statistics: ../data/dataset_statistics.csv

## Dataset Statistics

The code below generates descriptive statistics for the dataset.
It was obtained with AI assistance and adapted for this project.

The code produces:
- Data types for each column
- Missing value percentages
- Unique value counts
- Distribution by culture, country, period, and object type
- Output saved as a CSV file in the data/ folder

In [7]:
import os

# Check current working directory
print("Current folder:", os.getcwd())

# List all files here
print("\nFiles in current folder:")
for f in os.listdir():
    print(f)

Current folder: /Users/nandinikapoor/Documents/GitHub/My-first-Github-project/notebooks

Files in current folder:
Data obtained.ipynb
README.md
Met museum data.ipynb
.ipynb_checkpoints


In [9]:
import pandas as pd
import os

# Correct path — go up one level then into data folder
filename = "../data/asian_islamic_neareastern_met.csv"

df = pd.read_csv(filename, low_memory=False)

print("=== DATASET STATISTICS ===")
print(f"Rows: {len(df)}")
print(f"Columns: {len(df.columns)}")

print("\n--- Missing Values (%) ---")
print(df.isna().mean().round(2))

print("\n--- Unique Values per Column ---")
print(df.nunique())

print("\n--- Top Cultures ---")
print(df['Culture'].value_counts().head(10))

print("\n--- Top Countries ---")
print(df['Country'].value_counts().head(10))

print("\n--- Top Periods ---")
print(df['Period'].value_counts().head(10))

print("\n--- Object Types ---")
print(df['Object Name'].value_counts().head(10))

print("\n--- Public Domain ---")
print(df['Is Public Domain'].value_counts())

# Save statistics to data folder
missing_df = df.isna().mean().round(2).reset_index()
missing_df.columns = ['column', 'missing_percentage']
missing_df.to_csv('../data/dataset_statistics.csv', index=False)
print("\nStatistics saved to data/dataset_statistics.csv")

=== DATASET STATISTICS ===
Rows: 58796
Columns: 45

--- Missing Values (%) ---
Object Number              0.00
Is Highlight               0.00
Is Timeline Work           0.00
Is Public Domain           0.00
Object ID                  0.00
Gallery Number             0.94
Department                 0.00
AccessionYear              0.01
Object Name                0.00
Title                      0.49
Culture                    0.28
Period                     0.44
Dynasty                    1.00
Reign                      1.00
Portfolio                  1.00
Artist Role                0.82
Artist Display Name        0.82
Artist Display Bio         0.82
Artist Nationality         0.82
Artist Begin Date          0.82
Artist End Date            0.82
Artist Gender              0.99
Object Date                0.10
Object Begin Date          0.00
Object End Date            0.00
Medium                     0.00
Dimensions                 0.04
Credit Line                0.00
Geography Type           

## Dataset Statistics

### What this code does
This code loads the filtered MET Museum dataset and generates 
descriptive statistics to understand the structure, completeness, 
and content of the data.

### How I obtained the code
The code was developed with AI assistance (Claude by Anthropic) 
and adapted for this research project.

### What the code produces
- Total rows and columns
- Missing value percentages per column
- Unique value counts per column
- Distribution by culture, country, period, and object type
- Public domain status
- Output saved as `dataset_statistics.csv` in the data/ folder

### Key findings from the statistics
- **58,796 objects** across 3 departments
- **Japan (16,937) and China (13,502)** are the most represented cultures
- **Iran (6,243) and Egypt (3,505)** are the top countries of origin
- **Dynasty, Reign, County, Locale fields** are 100% empty — major gap
- **Artist information** is missing for 82% of objects
- **50,711 objects (86%)** are in the public domain
- Period data is dominated by Japanese Edo period — reflects collection bias